index.bs

```diff
@@ -424,18 +457,34 @@ For each <a>element proxy</a> for that element, perform the following steps:
     7. For each property in |inputProperties|, if the property’s computed value has changed,
         set the <a>animation requested flag</a> on the |animator| to <a>frame-requested</a>.
 
-The timelines present in an animator's <a>animator timeline list</a> can also affect the
-<a>animation requested flag</a>:
+Animator specifies a DocumentTimeline {#requesting-animation-frames-document-timeline}
+-------------------------------------
+
+If a {{DocumentTimelineOptions}} instance is present in an animator's <a>animator timeline options
+list</a> then that animator's <a>animation requested flag</a> <em>must</em> be set to
+<a>frame-requested</a> at the start of each animation frame.
+
+Animator specifies a ScrollTimeline {#requesting-animation-frames-scroll-timeline}
+-----------------------------------
+
+If a {{ScrollTimelineOptions}} |instance| is present in an animator's <a>animator timeline options
+list</a> then the user agent <em>must</em> run the following steps when the scroll position of the
+nearest non-visible overflow ancestor to the animator’s <a>root element</a> changes:
+
+  1. For each {{ScrollTimelineOptions}} instance |options| in the animator's <a>animator timeline
+       options list</a> do the following steps:
+
+    1. Let |timeline| be a {{ScrollTimeline}} created from |options|, adding the nearest non-visible
+         overflow ancestor of the <a>root element</a> as the {{scrollSource}}.
+
+    2. If the <a>current time of the ScrollTimeline</a> |timeline| would have changed due to the
+         change in scroll position, set the animator's <a>animation requested flag</a> to
+         <a>frame-requested</a>.
 
-    * If the string "document" is present in an animator's <a>animator timeline list</a>, then that
-        animator's <a>animation requested flag</a> must be set to <a>frame-requested</a> at the start
-        of each animation frame.
-    * If the string "scroll" is present in an animator's <a>animator timeline list</a>, then when
-        the scroll position of the nearest non-visible overflow ancestor to the animator’s
-        <a>root element</a> changes, that animator's <a>animation requested flag</a> must be set to
-        <a>frame-requested</a>.
+Issue: Creating a ScrollTimeline here is awkward.
```

Comment:
Agreed, we should construct the timelines once when the animator is instantiated and reuse them. Can you update the issue to be something like this.